test/parallel_api/ranges/std_ranges_transform.pass.cpp

```diff
@@ -0,0 +1,78 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "std_ranges_test.h"
+
+std::int32_t
+main()
+{
+#if _ENABLE_STD_RANGES_TESTING
+    using namespace test_std_ranges;
+    namespace dpl_ranges = oneapi::dpl::ranges;
+
+    //A checker below modifies a return type; a range based version with policy has another return type.
+    auto transform_unary_checker = [](std::ranges::random_access_range auto&& r_in,
+                                      std::ranges::random_access_range auto&& r_out, auto&&... args)
+    {
+        using Size = std::common_type_t<std::ranges::range_size_t<decltype(r_in)>,
+            std::ranges::range_size_t<decltype(r_out)>>;
+
+        Size size = std::ranges::size(r_out);
+        if constexpr(std::ranges::sized_range<decltype(r_in)>)
```

Comment:
actually, we have to use onedpl sized range logic

Reply:
I've removed the extra check for the transform unary case, because it expects only sized ranges. I introduced size range logic into the tests for the transform binary case.